### PR TITLE
arithmetic operation edge cases

### PR DESCRIPTION
**Pull request by mumencoder**

### CODE REVIEW

File: Content.Tests/DMProject/Tests/Expression/edge_arithmetic.dm [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	ASSERT( (-"str") == 0 )
+
+	ASSERT( (1.0 ** "abc") == 1 )
+	ASSERT( (1.0 ** null) == 1 )
+	ASSERT( (null ** "abc") == 1 )
+	ASSERT( (null ** 1.0) == 0 )
+
+	ASSERT( (null - null) == 0 )
+
+	ASSERT( (null * null) == 0 )
```
Comment (wixoaGit) [inline Content.Tests/DMProject/Tests/Expression/edge_arithmetic.dm:13]: This is a runtime error in DM v515: `runtime error: type mismatch: null ** 2`

This doesn't cause that however:
```
var/a = null
world.log << (a * null)
```

I think v515 made `null ** 2` an invalid operation and the compiler is turning `null * null` into that.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Comment by wixoaGit** [comment]
Closing in favor of #1488
